app/Libraries/User/FindForProfilePage.php

```diff
@@ -36,4 +36,27 @@ public static function find($id, ?string $type = null, ?bool $assertCanonicalId
 
         return $user;
     }
+
+    private static function fromAuth($id, ?string $type): ?User
+    {
+        $user = auth()->user();
+
+        if ($user === null) {
+            return null;
+        }
+
+        $userId = (string) $user->getKey();
+        switch ($type) {
+            case 'id':
+                $isSelf = $id === $userId;
+                break;
+            case 'username':
+                $isSelf = $id === $user->username;
+                break;
+            default:
+                $isSelf = $id === $userId || $id === $user->username;
```

Comment:
Wouldn't this match if someone else's username is the same as your id or something equally silly?